h/views/api/users.py

```diff
@@ -47,7 +47,7 @@ def create(request):
     try:
         user_unique_service.ensure_unique(appstruct, authority=client_authority_)
     except DuplicateUserError as err:
-        raise ConflictError(err)
+        raise ConflictError(str(err))
```

Comment:
Now that `ConflictError` is a Pyramid `HTTPError` subclass you can't do `ConflictError(err)` because you'd be passing `err` as the `detail` argument to `HTTPError` and the `detail` argument must be a string. You'll get a `TypeError: __str__ returned non-string (type DuplicateUserError)` when the `api_error()` view tries to render the JSON error message.

So use `str(err)` instead. `DuplicateUserError.__str__()` is the full error message that we want to return in the response (e.g. `"user with email address '...' already exists, user with username '...' already exists"`).

The end result is the same as on master: the full error message ends up in the `"reason"` field in the response.